src/util.cpp

```diff
@@ -37,13 +38,20 @@ std::string toHex(const std::vector<uint8_t> &data) {
     return result.str();
 }
 
-std::vector<uint8_t> randomBytes(size_t length) {
-    std::vector<uint8_t> buffer(length);
-    if (1 != RAND_bytes(buffer.data(), buffer.size())) {
-        // TODO: throw?
+std::vector<uint8_t> fromHex(const std::string &hexData) {
+    std::vector <uint8_t> binary;
+    for (size_t i=0; i<hexData.length(); i+=2) {
+        uint8_t b = (uint8_t) strtoul(hexData.substr(i,2).c_str(), NULL, 16);
```

Comment:
c_str() on temporary object is dangerous and can result in use after free

Reply:
fixed